### ISSUE TITLE
yii\bootstrap\Nav makes dropdown for sub items even if there is no subitems

### ISSUE THREAD

**Issue by bashibozuk** [yiisoft/yii2-bootstrap#68]
Code in view
```code
<?= \yii\bootstrap\Nav::widget([
            'items' => $this->context->menu
        ]); ?>

```

menu:
``` code
[
    0 => [
        'label' => 'Test'
        'url' => '/portalpage/page/default?user_id=13&id=1'
        'items' => []
        'options' => [
            'class' => 'menu-item'
            'target' => '_self'
        ]
    ]
    1 => [
        'label' => 'Ð—Ð° Ð½Ð°Ñ'
        'url' => '/portalpage/page/default?user_id=13&id=2'
        'items' => []
        'options' => [
            'class' => 'menu-item'
            'target' => '_self'
        ]
    ]
]
```
HTML
```html

<ul id="w2" class="nav">
<li class="menu-item dropdown open" target="_self"><a class="dropdown-toggle" href="/portalpage/page/default?user_id=13&amp;id=1" data-toggle="dropdown" aria-expanded="true">Test <b class="caret"></b></a>
<ul id="w3" class="dropdown-menu"></ul></li>
<li class="menu-item dropdown" target="_self"><a class="dropdown-toggle" href="/portalpage/page/default?user_id=13&amp;id=2" data-toggle="dropdown">За нас <b class="caret"></b></a><ul id="w4" class="dropdown-menu"></ul></li>
</ul>
```

As you can see there are redundant dropdowns for empty subitems array

I think that in  yii\bootstrap\Nav at line 185 the code should be 
``` code
 if (is_array($items) && !empty($items)) {
...
}
```
instead of
```code
 if (is_array($items)) {
....
}
```

**Comment by freezy-sk** [comment]
@samdark are you working on this? If not, I can do it

**Comment by samdark** [comment]
I'm not.

**Comment by freezy-sk** [comment]
@samdark I have one question. What should be expected behaviour?
a) Rendering nav item without carret as normal item
b) Rendering it with dropdown look, just without empty list?

**Comment by samdark** [comment]
Render it as normal item, I guess.

**Comment by bashibozuk** [comment]
I think that the whole code from line 179 to line 197 should look like this
``` code
	if (is_array($items) && !empty($items)) {
            $linkOptions['data-toggle'] = 'dropdown';
            Html::addCssClass($options, ['widget' => 'dropdown']);
            Html::addCssClass($linkOptions, ['widget' => 'dropdown-toggle']);
            if ($this->dropDownCaret !== '') {
                $label .= ' ' . $this->dropDownCaret;
            }
            
	    if ($this->activateItems) {
	        $items = $this->isChildActive($items, $active);
	    }
	    $items = $this->renderDropdown($items, $item);
            
        } else  {
	   $items = '';
	}

code

I think that looks much cleaner and you almost don't have if statement inside if statement which I hate :), but nevermind , I think that it should work your way too , but in case where you have not empty string for items instead of not empty array you will put the css classes for the dropdown and you will concatenate  the carret to the label, but still you will not render a submenu, so probably you will end with incorrect visualisation again.

**Comment by klimov-paul** [comment]
Resolved by #74

**Comment by yongshengli** [comment]
great
